### PR TITLE
pallet benchmark testing

### LINKED ISSUE #67: Benchmarks

### PR DESCRIPTION
**Pull request by suinuj**
resolves #67
The pr runs the benchmark test for all pallets in the parachain runtime and uses the result to replace default weights.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,7 +79,51 @@ generate-docker-compose-staging:
...
+	@./scripts/run-benchmark-pallet.sh pallet-democracy
+
+.PHONY: benchmark-pallet-collective
+benchmark-pallet-collective:
+	@./scripts/run-benchmark-pallet.sh pallet-collective
+
+.PHONY: benchmark-pallet-identity
+benchmark-pallet-identity:
+	@./scripts/run-benchmark-pallet.sh pallet-identity
+
```
Comment (Satoshi-Kusumoto) [inline Makefile]: obesolete

File: Makefile [hunk 2]
```diff
@@ -79,7 +79,51 @@ generate-docker-compose-staging:
...
+	@./scripts/run-benchmark-pallet.sh pallet-proxy
+
+.PHONY: benchmark-pallet-balances
+benchmark-pallet-balances:
+	@./scripts/run-benchmark-pallet.sh pallet-balances
+
+.PHONY: benchmark-pallet-assets
+benchmark-pallet-assets:
+	@./scripts/run-benchmark-pallet.sh pallet-assets
+
```
Comment (Satoshi-Kusumoto) [inline Makefile]: obsolete

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]
pls check the pallets included and the order
merge dev first

File: Makefile (hunk 1 above)
Comment (suinuj) [reply Makefile]: removed.

File: Makefile (hunk 2 above)
Comment (suinuj) [reply Makefile]: removed

File: runtime/Cargo.toml [hunk 3]
```diff
@@ -146,7 +146,16 @@ runtime-benchmarks = [
...
+	"xcm-builder/runtime-benchmarks",
 	"pallet-xcm/runtime-benchmarks",
 	"sp-runtime/runtime-benchmarks",
-	"xcm-builder/runtime-benchmarks",
+	"pallet-timestamp/runtime-benchmarks",
+	"pallet-scheduler/runtime-benchmarks",
+	"pallet-treasury/runtime-benchmarks",
+	"pallet-proxy/runtime-benchmarks",
+	'pallet-xcm/runtime-benchmarks',
+	"pallet-collator-selection/runtime-benchmarks",
```
Comment (Satoshi-Kusumoto) [inline runtime/Cargo.toml]: order

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -396,16 +397,16 @@ impl pallet_balances::Config for Runtime {
...
 	type ReserveIdentifier = [u8; 8];
 }
 
 impl pallet_utility::Config for Runtime {
-	type Event = Event;
-	type Call = Call;
-	type WeightInfo = pallet_utility::weights::SubstrateWeight<Runtime>;
+    type Event = Event;
+    type Call = Call;
+    type WeightInfo = weights::pallet_utility::WeightInfo<Runtime>;
```
Comment (Satoshi-Kusumoto) [inline runtime/src/lib.rs]: indents

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]
Generally LGTM
And what about membership
Now or the next PR?

**Comment by suinuj** [comment]
> Generally LGTM And what about membership Now or the next PR?
membership included.

File: runtime/Cargo.toml [hunk 5]
```diff
@@ -146,7 +146,17 @@ runtime-benchmarks = [
...
 	"pallet-xcm/runtime-benchmarks",
 	"sp-runtime/runtime-benchmarks",
 	"xcm-builder/runtime-benchmarks",
+	"pallet-timestamp/runtime-benchmarks",
+	"pallet-scheduler/runtime-benchmarks",
+	"pallet-treasury/runtime-benchmarks",
+	"pallet-proxy/runtime-benchmarks",
+	'pallet-xcm/runtime-benchmarks',
+	"pallet-collator-selection/runtime-benchmarks",
+	"pallet-membership/runtime-benchmarks"
```
Comment (Satoshi-Kusumoto) [inline runtime/Cargo.toml]: order!

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]
1. Pls carefully check which pallets are involved. Makefile is obsolete and say, Multisig is not included
2. order. Do not ignore review comments

File: runtime/Cargo.toml (hunk 5 above)
Comment (suinuj) [reply runtime/Cargo.toml]: updated.

File: Makefile [hunk 6]
```diff
@@ -107,7 +107,51 @@ format:
...
+.PHONY: benchmark-pallet-membership
+benchmark-pallet-membership:
+	@./scripts/run-benchmark-pallet.sh pallet-membership
+
+.PHONY: benchmark-pallet-collator-selection
+benchmark-pallet-collator-selection:
+	@./scripts/run-benchmark-pallet.sh pallet-collator-selection
+
+.PHONY: benchmark-pallet-multisig
```
Comment (Satoshi-Kusumoto) [inline Makefile:152]: Still have account-linker, ocw, nft

File: node/src/chain_spec.rs [hunk 7]
```diff
@@ -92,7 +92,7 @@ pub fn default_parachain_properties() -> Option<Properties> {
 	parachain_properties("LIT", 12, 31)
 }
 
-const DEFAULT_CANDIDACY_BOND: Balance = 16 * UNIT;
+const DEFAULT_CANDIDACY_BOND: Balance = 1;
```
Comment (Satoshi-Kusumoto) [inline node/src/chain_spec.rs]: should not modify actual code
modify tests or setup pls

File: runtime/Cargo.toml [hunk 8]
```diff
@@ -150,4 +150,15 @@ runtime-benchmarks = [
...
+	"pallet-democracy/runtime-benchmarks",
+	"pallet-balances/runtime-benchmarks",
+	"pallet-utility/runtime-benchmarks",
+	"pallet-scheduler/runtime-benchmarks",
+	"pallet-treasury/runtime-benchmarks",
+	"pallet-proxy/runtime-benchmarks",
+	'pallet-xcm/runtime-benchmarks',
+	"pallet-collator-selection/runtime-benchmarks",
+	"pallet-membership/runtime-benchmarks",
+	"pallet-multisig/runtime-benchmarks"
```
Comment (Satoshi-Kusumoto) [inline runtime/Cargo.toml]: alphabetic order for the whol toml

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]

File: node/src/chain_spec.rs (hunk 7 above)
Comment (suinuj) [reply node/src/chain_spec.rs]: test code from upstream, I discussed with Kailai, the constant just impacts the dev environemnt.

File: Makefile (hunk 6 above)
Comment (suinuj) [reply Makefile:152]: removed.

File: runtime/Cargo.toml (hunk 8 above)
Comment (suinuj) [reply runtime/Cargo.toml]: updated.

**Review by Satoshi-Kusumoto — APPROVED** [review]

**Pull request merged**